tfhe/src/core_crypto/algorithms/polynomial_algorithms.rs

```diff
@@ -356,24 +367,90 @@ pub fn polynomial_wrapping_monic_monomial_mul<Scalar, OutputCont, InputCont>(
     let polynomial_size = output.polynomial_size().0;
     let remaining_degree = monomial_degree.0 % polynomial_size;
 
-    for (dst, &src) in output[..remaining_degree]
-        .iter_mut()
-        .zip(input[polynomial_size - remaining_degree..].iter())
-    {
-        *dst = src.wrapping_neg();
+    let full_cycles_count = monomial_degree.0 / polynomial_size;
+    if full_cycles_count % 2 == 0 {
+        copy_with_neg(
+            &mut output[..remaining_degree],
+            &input[polynomial_size - remaining_degree..],
+        );
+        copy_without_neg(
+            &mut output[remaining_degree..],
+            &input[..polynomial_size - remaining_degree],
+        );
+    } else {
+        copy_without_neg(
+            &mut output[..remaining_degree],
+            &input[polynomial_size - remaining_degree..],
+        );
+        copy_with_neg(
+            &mut output[remaining_degree..],
+            &input[..polynomial_size - remaining_degree],
+        );
     }
+}
 
-    let dst_slice = &mut output[remaining_degree..];
-    let dst_slice_len = dst_slice.len();
-    let src_slice = &input[..dst_slice_len];
-    dst_slice.copy_from_slice(src_slice);
+pub(crate) fn polynomial_wrapping_monic_monomial_mul_and_subtract<Scalar, OutputCont, InputCont>(
```

Comment:
could we have a small docstring explaining what the operation does ? :) a bit like you described in the external product with a small example here would be very nice